programs/vote/src/vote_state/mod.rs

```diff
@@ -191,24 +191,56 @@ fn set_vote_account_state(
     }
 }
 
-fn check_update_vote_state_slots_are_valid(
+fn check_and_filter_vote_state_update(
     vote_state: &VoteState,
     vote_state_update: &mut VoteStateUpdate,
     slot_hashes: &[(Slot, Hash)],
 ) -> Result<(), VoteError> {
-    if vote_state_update.lockouts.is_empty() {
+    check_and_filter_proposed_vote_state(
+        vote_state,
+        &mut vote_state_update.lockouts,
+        &mut vote_state_update.root,
+        vote_state_update.hash,
+        slot_hashes,
+    )
+}
+
+fn check_and_filter_tower_sync(
+    vote_state: &VoteState,
+    tower_sync: &mut TowerSync,
+    slot_hashes: &[(Slot, Hash)],
+) -> Result<(), VoteError> {
+    check_and_filter_proposed_vote_state(
+        vote_state,
+        &mut tower_sync.lockouts,
+        &mut tower_sync.root,
+        tower_sync.hash,
+        slot_hashes,
+    )
+}
```

Comment:
nit, but I think we can just directly call `check_and_filter_proposed_vote_state` from both here, no need for this thin wrapper

Reply:
thin wrapper was for easy migration of the tests, will clean it up in the next pr for the rest of the test migration + deprecation of VoteStateUpdate